src/classLibrary/ParsingStrategies/IfElseResolver.ts

```diff
@@ -0,0 +1,210 @@
+import {Utility} from "../Utility/Utility";
+import {FileSystemInstance, 
+	FileContent,
+	VirtualFileSystemInstance,
+	IParsingStrategy,
+	Syntax,
+	CyanSafe,
+	} from "../interfaces/interfaces";
+
+class IfElseResolver implements IParsingStrategy {
+
+	private util: Utility;
+	
+	constructor(util: Utility) {
+		this.util = util;
+	}
+	
+	Count(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): Map<string, number> {
+    	const flags: string[] = this.util.FlattenObject(cyan.flags).Keys();
+        const result: Map<string, number> = new Map<string, number>();
+        const syntaxes: Syntax[] = cyan.syntax;
+
+        virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            flags.Each((flag: string) => {
+                const allPossibleIfs = this.ModifyIfWithAllSyntax(flag, syntaxes).concat(this.ModifyInverseIfWithAllSyntax(flag, syntaxes));
+                allPossibleIfs.Each((key: string) => {
+                    const count = this.CountKeyInVFS(key, virtualFile)
+									.AtMax(this.CountKeyInVFS(this.ConvertIfToEndKeyword(key), virtualFile));
+                    if (result.has(flag)) {
+                        result.set(flag, result.get(flag)! + count);
+                    } else {
+                        result.set(flag, count);
+                    }
+                });
+            });
+        });
+
+        return result;
+  	}
+
+	private CountKeyInVFS(key: string, virtualFile: VirtualFileSystemInstance): number {
+		return VirtualFileSystemInstance.match(virtualFile, {
+			File: (file: FileSystemInstance) => {
+				let tempCount = 0;
+				if (file["content"] != null && file.ignore.ifElseResolver.content) {
+					FileContent.if.String(file.content, (str) => {
+						tempCount += str.Count(key);
+					});
+				}
+				return tempCount;
+			},
+			default: () => 0,
+		});
+	}
+
+	ModifyIfWithAllSyntax(v: string, syntaxes: Syntax[]): string[] {
+		const allPossibleIfs: string[] = [];
+		syntaxes.Each(syntax => {
+			allPossibleIfs.push(`if${syntax[0] + v + syntax[1]}`);
+		})
+		return allPossibleIfs;
+	}
+
+	ModifyInverseIfWithAllSyntax(v: string, syntaxes: Syntax[]): string[] {
+		const allPossibleInverseIfs: string[] = [];
+		syntaxes.Each(syntax => {
+			allPossibleInverseIfs.push(`if!${syntax[0] + v + syntax[1]}`);
+		})
+		return allPossibleInverseIfs;
+	}
+
+	ConvertIfToEndKeyword(ifStatement: string): string
+	{
+		return ifStatement.replace('if', 'end');
+	}
+
+	
+	ResolveFiles(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+		return virtualFiles;
+	};
+
+	ResolveContents(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+		const flagsMap: Map<string, boolean> = this.util.FlattenObject(cyan.flags).MapValue((boolString: string) => boolString == "true");
```

Comment:
I just realized this, I think in `cyan.flags`, it will be an Object with boolean values instead of string values though. I think it will make everything more clear.

```
{
    a: true,
    b: {
        c: true,
        d: {
            e: false,
            f: false
        }
    },
    g: true
};
```